operator/pkg/test/environment/fakekubeclient.go

```diff
@@ -0,0 +1,40 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package environment
+
+import (
+	"context"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FakeKubeClient struct {
```

Comment:
Trying to understand why you need this at all. What's the special bits about the string suffix? Maybe we can discuss offline?

Reply:
Master has a dependency on control plane service to be created and loadbalancer hostname to be populated before it can generate certs. In tests, I need to make sure hostname gets populated.
Let's discuss this more offline.